templates/project/.github/workflows/publish.yml

```diff
@@ -16,20 +21,26 @@ jobs:
       uses: actions/checkout@v3
       with:
         fetch-depth: 2
+    - name: Set Project Name
+      id: set_project_name
+      run: |
+          echo "PROJECT_NAME=`cargo metadata --no-deps --format-version=1 | jq '.packages[0].name' -r`" >> $GITHUB_OUTPUT
```

Comment:
oh neat

Reply:
Unfortunately, we don't have `jq` in our canon images so I'm going to need to do that before we can merge this.